File: Leet 3043/solution.go
package main

import (
	"fmt"
	"math"
)

func maxInt(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func intLength(num int) int {
	if num == 0 {
		return 1
	}

	abs := math.Abs(float64(num))
	log := math.Log10(abs)
	floor := math.Floor(log)

	return int(floor) + 1
}

func longestCommonPrefix(arr1 []int, arr2 []int) int {
	seen := make(map[int]bool)

	for i := range arr1 {
		num := arr1[i]

		for num != 0 && !seen[num] {
			seen[num] = true
			num /= 10
		}
	}

	var result int = 0

	for i := range arr2 {
		num := arr2[i]

		for num != 0 && !seen[num] {
			num /= 10
		}

		if num != 0 {
			result = maxInt(result, intLength(num))
		}
	}

	return result
}

func main() {
	fmt.Println("3043. Find the Length of the Longest Common Prefix")

	arr1 := []int{1, 10, 100}
	arr2 := []int{1000}

	var result int = longestCommonPrefix(arr1, arr2)
	fmt.Printf("Longest common prefix for [1 10 100] and [1000] is %d\n", result)
}

File: Leet 3043/solution.cs
public class Solution {
    private static int IntLength(int number) {
        if(number == 0)
        {
            return 1;
        }
        
        var abs = Math.Abs(number);
        var log = Math.Log10(abs);
        var floor = Math.Floor(log);
        return (int)floor + 1;
    }

    public int LongestCommonPrefix(int[] arr1, int[] arr2) {
        HashSet<int> seen = [];

        foreach(int number in arr1)
        {
            int current = number;
            while(current != 0 && !seen.Contains(current))
            {
                seen.Add(current);
                current /= 10;
            }
        }

        int result = 0;
        foreach(int number in arr2)
        {
            int current = number;
            while (current != 0 && !seen.Contains(current))
            {
                current /= 10;
            }

            if (current != 0)
            {
                result = Math.Max(result, IntLength(current));
            }
        }

        return result;
    }
}

class Program
{
    private static void Main()
    {
        Console.WriteLine("3043. Find the Length of the Longest Common Prefix");

        int[] arr1 = [1,10,100];
        int[] arr2 = [1000];

        Solution solution = new();
        int result = solution.LongestCommonPrefix(arr1, arr2);
        Console.WriteLine($"Longest common prefix between [1 10 100] and [1000] is {result}");
    }
}